spec/controllers/idv/image_uploads_controller_spec.rb

```diff
@@ -56,34 +56,21 @@
         stub_analytics
         stub_attempts_tracker
 
-        expect(@analytics).to receive(:track_event).with(
-          'IdV: doc auth image upload form submitted,',
-          success: false,
-          errors: {
-            front: ['Please fill in this field.'],
-          },
-          error_details: {
-            front: { blank: true },
-          },
-          user_id: user.uuid,
-          submit_attempts: 1,
-          remaining_submit_attempts: IdentityConfig.store.doc_auth_max_attempts - 1,
-          pii_like_keypaths: pii_like_keypaths,
-          flow_path: 'standard',
-        ).exactly(0).times
-
-        expect(@analytics).not_to receive(:track_event).with(
-          'IdV: doc auth image upload vendor submitted',
-          any_args,
-        )
-
         expect(@irs_attempts_api_tracker).to receive(:track_event).with(
           :idv_document_upload_submitted,
           any_args,
         )
 
         action
 
+        expect(@analytics).not_to have_logged_event(
```

Comment:
Do you mind clarifying why `have_logged_event` is better or more useful than `to receive(:track_event).with(...)`? Mostly curious and would be good to know for future tests.

Reply:
Not sure it's Matt's reason, but one reason to prefer it is that `receive(:track_event)` may override our custom [PII alerting](https://github.com/18F/identity-idp/blob/439fa72f3c545e77088c80d17d80499af238f825/spec/support/fake_analytics.rb#L7-L8) and [documentation enforcing](https://github.com/18F/identity-idp/blob/439fa72f3c545e77088c80d17d80499af238f825/spec/support/fake_analytics.rb#L74-L79) helpers.

Reply:
Yes, it is mostly those things. Also consistent use of `have_logged_event` gives us some freedom to make changes to how we test events in a single place rather than updating every single callsite